eng/scripts/Language-Settings.ps1

```diff
@@ -5,6 +5,58 @@ $packagePattern = "*.zip"
 $MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/python-packages.csv"
 $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=python%2F&delimiter=%2F"
 
+function Get-AllPackageInfoFromRepo ($serviceDirectory)
+{
+  $allPackageProps = @()
+  $searchPath = "sdk/*/*/setup.py"
+  if ($serviceDirectory)
+  {
+    $searchPath = "sdk/${serviceDirectory}/*/setup.py"
+  }
+  Push-Location $RepoRoot
+  $allSetupProps = $null
+  try
+  {
+    pip install packaging==20.4 -q -I
+    $allSetupProps = python "eng\scripts\get_package_properties.py" -s $searchPath
+  }
+  catch
+  {
+    # This is soft error and failure is expected for python metapackages
+    LogError "Failed to get all package properties"
+  }
+  Pop-Location
```

Comment:
If this is only for the python script run then put the push in the try block and put the pop in a finally block.